src/zos.cc

```diff
@@ -1508,35 +1508,37 @@ extern "C" int execvpe(const char *name, char *const argv[],
   int lp, ln;
   const char *p;
   int eacces = 0, etxtbsy = 0;
-  char *bp, *cur;
+  char *dp = NULL, *cur;
 
   // Get the path we're searching
   int len;
-  if (cur = getenv("PATH")) {
+  if ((cur = getenv("PATH")) != NULL) {
     len = strlen(cur);
-  }
-  else {
+  } else {
     // If PATH is not defined, get the default from confstr
     len = confstr(_CS_PATH, NULL, 0);
     if (len) {
-       char *dp = (char*)alloca (len);
-       confstr (_CS_PATH, dp, len);
-       cur = dp;
-    } else { len = 1; }
+       dp = (char*)malloc(len + 1);
+       if (dp == NULL)
+         return errno ? errno : ENOMEM;
+    } else
+       len = 1;
   }
   char path[len + 1];
```

Comment:
I'd have to look at the code closer, but you are allocating 3 buffers in addition to the original.  That is unusual.  Can you refine this code?

Reply:
Both path and buf char arrays were also in the original code; what's the 3rd bufffer? path needs to store either dp (confstr()) or cur (getenv()), and buf stores the updated path; cur and dp pointers need to be separate since cur would not be freed while dp would.

I did however remove use of char *bp in next commit.

Reply:
The buffers are dp, path and buf.  buf and path are discarded when the function returns.

Reply:
Yeah, and dp (that was previously alloca'd) is freed.